src/components/Form/components/CardHolderNameInput.tsx

```diff
@@ -1,14 +1,13 @@
-import { ChangeEvent, Dispatch } from "react";
+import { ChangeEvent } from "react";
 import { EFormaActions, EFormData } from "../../../enums/Form";
-import { FormActions } from "../../../reducers/form-reducer";
+import { useForm } from "../../../hooks/useForm";
 
-type CardholderNameInputProps = {
-  state: string;
-  dispatch: Dispatch<FormActions>;
-  error?: string;
-};
+const CardHolderNameInput = () => {
+
+  const { dispatch, state, errors } = useForm()
+  const {cardholderName} = state
+  const cardholderNameError = errors[EFormData.CardholderName];
 
-const CardHolderNameInput = ({ state, dispatch, error }: CardholderNameInputProps) => {
   const handleChange = (
```

Comment:
this handleChange should be ideal to have it in the context so you remove most of the logic to the context and keep clean the component